src/gen-ts.ts

```diff
@@ -297,9 +299,34 @@ function handleMethods(analyzerMethods: Iterable<analyzer.Method>):
     });
     m.description = method.description || '';
 
-    for (const param of method.params || []) {
-      const {type, optional, rest} = closureParamToTypeScript(param.type);
-      m.params.push(new ts.Param({name: param.name, type, optional, rest}));
+    let requiredAhead = false;
+    for (let i = (method.params || []).length - 1; i >= 0; i--) {
```

Comment:
Arg, complicated three arg for loop. What about:

```typescript
function* reversedIter<T>(arr: T[]) {
  for (let i = arr.length - 1; i >= 0; i--) {
    yield arr[i];
  }
}
```

Then iteration can be:

```typescript
for (const param of reversed(method.params || [])) {

}
```

Reply:
Done.